### PR TITLE
Add distributor bucket db memory usage metrics

### PR DESCRIPTION
**Pull request by vekterli**
@geirst please review.

One question up for debate is the metric dimension key name of the bucket DB metrics. There is one dimension for the mutable DBs and one for the read-only DBs. I've just gone with "type" as the key here, but we might have something more suitable.

### CODE REVIEW

File: metrics/src/vespa/metrics/common/memory_usage_metrics.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+#include "memory_usage_metrics.h"
+#include <vespa/vespalib/util/memoryusage.h>
+
+namespace metrics {
+
+MemoryUsageMetrics::MemoryUsageMetrics(metrics::MetricSet* parent)
+    : MetricSet("memory_usage", {}, "The memory usage for a given component", parent),
+      _allocated_bytes("allocated_bytes", {}, "The number of allocated bytes", this),
+      _used_bytes("used_bytes", {}, "The number of used bytes (<= allocatedbytes)", this),
```
Comment (geirst) [inline metrics/src/vespa/metrics/common/memory_usage_metrics.cpp:11]: Consider changing description to match metric name: "<= allocated_bytes"

File: metrics/src/vespa/metrics/common/memory_usage_metrics.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      _dead_bytes("dead_bytes", {}, "The number of dead bytes (<= usedbytes)", this),
```
Comment (geirst) [inline metrics/src/vespa/metrics/common/memory_usage_metrics.cpp:12]: Consider changing description to match metric name: "<= used_bytes"

File: storage/src/vespa/storage/distributor/distributormetricsset.cpp [hunk 3]
```diff
@@ -2,11 +2,19 @@
...
 #include <vespa/metrics/loadmetric.hpp>
 #include <vespa/metrics/summetric.hpp>
+#include <vespa/vespalib/util/memoryusage.h>
 
 namespace storage::distributor {
 
 using metrics::MetricSet;
 
+BucketDbMetrics::BucketDbMetrics(const vespalib::string& db_type, metrics::MetricSet* owner)
+    : metrics::MetricSet("bucket_db", {{"type", db_type}}, "", owner),
```
Comment (geirst) [inline storage/src/vespa/storage/distributor/distributormetricsset.cpp:12]: I think "type" could interfere with other dimension names. Consider "bucket_db_type" instead?

**Review by geirst — APPROVED** [review]
👍 Some comments to consider, especially the dimension name for bucket db as you already pointed out.

File: metrics/src/vespa/metrics/common/memory_usage_metrics.cpp (hunk 1 above)
Comment (vekterli) [reply metrics/src/vespa/metrics/common/memory_usage_metrics.cpp]: Done

File: storage/src/vespa/storage/distributor/distributormetricsset.cpp (hunk 3 above)
Comment (vekterli) [reply storage/src/vespa/storage/distributor/distributormetricsset.cpp]: Done

File: metrics/src/vespa/metrics/common/memory_usage_metrics.cpp (hunk 2 above)
Comment (vekterli) [reply metrics/src/vespa/metrics/common/memory_usage_metrics.cpp]: Done

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**